packages/library-base/src/commonMain/kotlin/io/realm/Queryable.kt

```diff
@@ -25,6 +25,6 @@ package io.realm
 /**
  * Interface holding common query methods.
  */
-interface Queryable<T : RealmObject> {
+interface Queryable<T> {
     fun query(query: String = "TRUEPREDICATE", vararg args: Any?): RealmResults<T>
```

Comment:
Not sure it makes sense that this have a default value. It would only allow patterns like `realm.query(Sample::class, "id == 42").query().find` which doesn't seem useful?

Reply:
Remember this is the "old" query API and it's only implemented by `RealmResults` - notice it returns `RealmResults` too. The new API has a default `TRUEPREDICATE` parameter as mentioned above.